### PR TITLE
[821] Make it clearer as to why a deleted job posting is no longer available on the service

### PR DESCRIPTION
**Pull request by erbridge**
## Trello card URL:

https://trello.com/c/fBHon4UL/821-make-it-clearer-as-to-why-a-deleted-job-posting-is-no-longer-available-on-the-service

## Changes in this PR:

Add a new error view with more specific copy if the vacancy the user is trying to get to was trashed.

### Before

Screen Shot 2019-04-17 at 14 22 10

### After

Screen Shot 2019-04-17 at 14 22 24

### CODE REVIEW

**Comment by erbridge** [comment]
[Make it clearer as to why a deleted job posting is no longer available on the service](https://trello.com/c/fBHon4UL/821-make-it-clearer-as-to-why-a-deleted-job-posting-is-no-longer-available-on-the-service)

File: app/controllers/vacancies_controller.rb [hunk 1]
```diff
@@ -24,7 +24,14 @@ def index
   end
 
   def show
-    vacancy = Vacancy.listed.friendly.find(id)
+    begin
+      vacancy = Vacancy.listed.friendly.find(id)
+    rescue ActiveRecord::RecordNotFound
+      raise unless Vacancy.trashed.friendly.exists?(id)
```
Comment (pezholio) [inline app/controllers/vacancies_controller.rb]: I like the implementation, but can we get some tests for this?

Comment (pezholio) [reply app/controllers/vacancies_controller.rb]: You can probably use Rspec's [render template matcher](https://relishapp.com/rspec/rspec-rails/docs/matchers/render-template-matcher) in a controller test for this

File: app/controllers/vacancies_controller.rb [hunk 2]
```diff
@@ -24,7 +24,14 @@ def index
...
+
+      return render '/errors/trashed_vacancy_found'
```
Comment (pezholio) [inline app/controllers/vacancies_controller.rb]: Also, can you add `status: :not_found` here too, so the server returns the correct status code?

Comment (erbridge) [reply app/controllers/vacancies_controller.rb]: :+1:

**Pull request merged**